### PR TITLE
[CFE-400]: Feature: add publish/republish tag to modules table

### PR DESCRIPTION
**Pull request by Poafs1**

### CODE REVIEW

File: src/lib/components/table/modules/ModulesTableMobileCard.tsx [hunk 1]
```diff
@@ -43,14 +43,26 @@ export const ModulesTableMobileCard = ({
...
+              <MobileLabel label="owner" />
+              <ExplorerLink
+                value={moduleInfo.address}
+                type="user_address"
+                showCopyOnHover
+              />
+            </Flex>
+            <Flex direction="column" gap={1}>
+              <MobileLabel label="Action" />
+              <Box>
```
Comment (songwongtp) [inline src/lib/components/table/modules/ModulesTableMobileCard.tsx:57]: do we need `Box`?

File: src/lib/components/table/modules/ModulesTableHeader.tsx [hunk 2]
```diff
@@ -20,6 +20,7 @@ export const ModulesTableHeader = ({
       </TableHeader>
     )}
     <TableHeader>Creator</TableHeader>
+    {!isPublishedModules && <TableHeader>Publishing Activity</TableHeader>}
     {!isPublishedModules && <TableHeader>Timestamp</TableHeader>}
```
Comment (songwongtp) [inline src/lib/components/table/modules/ModulesTableHeader.tsx:24]: combine

File: src/lib/components/table/modules/ModulesTableRow.tsx [hunk 3]
```diff
@@ -73,6 +75,11 @@ export const ModulesTableRow = ({
...
           />
         </TableRow>
+        <TableRow>
+          <Tag variant={isRepublished ? "primary-light" : "gray"}>
+            {isRepublished ? "Republish" : "Publish"}
+          </Tag>
+        </TableRow>
         {!isPublishedModules && moduleInfo.latestUpdated && (
           <TableRow>
             <Flex direction="column" gap={1}>
```
Comment (songwongtp) [inline src/lib/components/table/modules/ModulesTableRow.tsx:85]:
```
{!isPublishedModules && (
    <>
        <TableRow>....</TableRow>
        <TableRow>
            ...
            <Text ...>
                {moduleInfo.latestUpdated ? `(${dateFromNow(moduleInfo.latestUpdated)})` : "N/A"}
            </Text>
        </TableRow>
    </>
)}
```

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/components/table/modules/ModulesTable.tsx [hunk 4]
```diff
@@ -47,6 +47,7 @@ export const ModulesTable = ({
           moduleInfo={module}
           templateColumns={templateColumns}
           isPublishedModules={isPublishedModules}
+          isRepublished={!!module.isRepublished}
```
Comment (songwongtp) [inline src/lib/components/table/modules/ModulesTable.tsx:50]: remove use directly from `moduleInfo`

File: src/lib/components/table/modules/ModulesTableRow.tsx [hunk 5]
```diff
@@ -73,13 +75,26 @@ export const ModulesTableRow = ({
             showCopyOnHover
           />
         </TableRow>
-        {!isPublishedModules && moduleInfo.latestUpdated && (
+        <TableRow>
+          <Tag variant={isRepublished ? "primary-light" : "gray"}>
+            {isRepublished ? "Republish" : "Publish"}
+          </Tag>
+        </TableRow>
```
Comment (songwongtp) [inline src/lib/components/table/modules/ModulesTableRow.tsx:82]: ditto

File: src/lib/components/table/modules/ModulesTableMobileCard.tsx [hunk 6]
```diff
@@ -43,14 +43,25 @@ export const ModulesTableMobileCard = ({
...
+            </Flex>
+            <Flex direction="column" gap={1}>
+              <MobileLabel label="Action" />
+              <Tag
+                width="fit-content"
+                variant={moduleInfo.isRepublished ? "primary-light" : "gray"}
+              >
+                {moduleInfo.isRepublished ? "Republish" : "Publish"}
+              </Tag>
+            </Flex>
```
Comment (songwongtp) [inline src/lib/components/table/modules/ModulesTableMobileCard.tsx:63]: not show in my published module

**Review by songwongtp — CHANGES_REQUESTED** [review]

**Review by songwongtp — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**